rir/src/compiler/pir/instruction.h

```diff
@@ -119,6 +119,14 @@ enum class Controlflow : uint8_t {
     Branch,
 };
 
+// How an instruction modifies visibility
+enum class VisibilityMod : uint8_t {
```

Comment:
Those names are really hard to read. Also they repeat and I don't understand why we need the 4th option. How about this:

```
enum class VisibilityFlag : uint8_t {
  Set,
  Clear,
  Unknown
};
```

also, maybe make this an inner class of Instruction

Reply:
We have a lot of other types which are outside of instruction, so I don't think this should be an inner class unless we move all those types inside too?

Reply:
fair enough. But `VisibilityFlag::Visible` is not a good name. It does not make sense to use scoped enum, and then repeat the scope name again. That's like `Door::doorOpen`, when you can say `Door::open`

Reply:
ok, fixed (should've put in the commit message)